docs/working-with-kyverno.md

```diff
@@ -0,0 +1,314 @@
+   kind: Service
+   metadata:
+     labels:
+       app: kyverno
+       app.kubernetes.io/component: kyverno
+       app.kubernetes.io/instance: kyverno
+       app.kubernetes.io/name: kyverno
+       app.kubernetes.io/part-of: kyverno
+       app.kubernetes.io/version: latest
+     name: kyverno-svc
+     namespace: kyverno
+   spec:
+     type: NodePort
+     ports:
+     - name: https
+       port: 443
+       targetPort: https
+       nodePort: {{ nodePort }}
+     selector:
+       app: kyverno
+       app.kubernetes.io/name: kyverno
+   ---
+   apiVersion: v1
+   kind: Service
+   metadata:
+     labels:
+       app: kyverno
+       app.kubernetes.io/component: kyverno
+       app.kubernetes.io/instance: kyverno
+       app.kubernetes.io/name: kyverno
+       app.kubernetes.io/part-of: kyverno
+       app.kubernetes.io/version: latest
+     name: kyverno-svc-metrics
+     namespace: kyverno
+   spec:
+     ports:
+     - name: metrics-port
+       port: 8000
+       targetPort: metrics-port
+     selector:
+       app: kyverno
+       app.kubernetes.io/name: kyverno
+   ---
+   apiVersion: apps/v1
+   kind: Deployment
+   metadata:
+     labels:
+       app: kyverno
+       app.kubernetes.io/component: kyverno
+       app.kubernetes.io/instance: kyverno
+       app.kubernetes.io/name: kyverno
+       app.kubernetes.io/part-of: kyverno
+       app.kubernetes.io/version: latest
+     name: kyverno
+     namespace: kyverno
+   spec:
+     replicas: 1
+     selector:
+       matchLabels:
+         app: kyverno
+         app.kubernetes.io/name: kyverno
+     strategy:
+       rollingUpdate:
+         maxSurge: 1
+         maxUnavailable: 40%
+       type: RollingUpdate
+     template:
+       metadata:
+         labels:
+           app: kyverno
+           app.kubernetes.io/component: kyverno
+           app.kubernetes.io/instance: kyverno
+           app.kubernetes.io/name: kyverno
+           app.kubernetes.io/part-of: kyverno
+           app.kubernetes.io/version: latest
+       spec:
+         affinity:
+           podAntiAffinity:
+             preferredDuringSchedulingIgnoredDuringExecution:
+             - podAffinityTerm:
+                 labelSelector:
+                   matchExpressions:
+                   - key: app.kubernetes.io/name
+                     operator: In
+                     values:
+                     - kyverno
+                 topologyKey: kubernetes.io/hostname
+               weight: 1
+         containers:
+         - args:
+           - --filterK8sResources=[Event,*,*][*,kube-system,*][*,kube-public,*][*,kube-node-lease,*][Node,*,*][APIService,*,*][TokenReview,*,*][SubjectAccessReview,*,*][*,kyverno,kyverno*][Binding,*,*][ReplicaSet,*,*][ReportChangeRequest,*,*][ClusterReportChangeRequest,*,*][PolicyReport,*,*][ClusterPolicyReport,*,*]
+           - -v=2
+           - --autogenInternals=false
+           - --kubeconfig=/etc/kubeconfig
+           - --serverIP={{ nodeIP }}:{{ nodePort }}
```

Comment:
I guess `kyverno` will bind the `nodeIP` and listen on the `nodePort`, is that right?

What's the default value of `--serverIP`?

Reply:
yeah，serverIP is the address of the webhook server. When deployed in the cluster, serverIP is empty.

Reply:
empty means binds on `0.0.0.0`?

Reply:
Will not have this config. Actually, kyverno has some logic that check this config,  use url clientconfig when serverIP is not "" and use service clientconfig when serverIP is "" which means kyverno is deployed in cluster.